http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcRequestAuthenticator.java

```diff
@@ -416,4 +475,104 @@ private static boolean hasScope(String scopeParam, String targetScope) {
         }
         return false;
     }
+    private String convertToRequestParameter(URIBuilder redirectUriBuilder, String redirectUri, String state, List<NameValuePair> forwardedQueryParams) throws Exception {
+        redirectUriBuilder.addParameter(SCOPE, OIDC_SCOPE);
+        forwardedQueryParams.add(new BasicNameValuePair(STATE, state));
+        forwardedQueryParams.add(new BasicNameValuePair(REDIRECT_URI, redirectUri));
+        forwardedQueryParams.add(new BasicNameValuePair(RESPONSE_TYPE, CODE));
+        forwardedQueryParams.add(new BasicNameValuePair(CLIENT_ID, deployment.getResourceName()));
+
+        JwtClaims jwtClaims = new JwtClaims();
+        jwtClaims.setIssuer(deployment.getIssuerUrl());
+        jwtClaims.setAudience(deployment.getProviderUrl());
+        for ( NameValuePair parameter: forwardedQueryParams) {
+            jwtClaims.setClaim(parameter.getName(), parameter.getValue());
+        }
+
+        // sign JWT first before encrypting
+        JsonWebSignature signedRequest = signRequest(jwtClaims);
+
+        // Encrypting optional
+        if (deployment.getRequestEncryptAlgorithm() != null && !deployment.getRequestEncryptAlgorithm().isEmpty() &&
+            deployment.getRequestEncryptEncValue() != null && !deployment.getRequestEncryptEncValue().isEmpty()) {
+            return encryptRequest(signedRequest).getCompactSerialization();
+        } else {
+            return signedRequest.getCompactSerialization();
+        }
+    }
+
+    public KeyPair getkeyPair() {
+        if (deployment.getClientKeyStore().contains(PROTOCOL_CLASSPATH)) {
+            deployment.setClientKeyStore(deployment.getClientKeyStore().replace(PROTOCOL_CLASSPATH, Objects.requireNonNull(Thread.currentThread().getContextClassLoader().getResource("")).getPath()));
+        }
+        if (!deployment.getRequestSignatureAlgorithm().equals(NONE) && deployment.getClientKeyStore() == null){
+            throw new RuntimeException("Invalid keystore configuration for signing Request Objects.");
```

Comment:
@PrarthonaPaul Just a minor, please put this exception into the ElytronMessages class